src/main/java/core/basesyntax/strategy/impl/BalanceOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(Storage storage, FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        if (quantity < 0) {
+            throw new RuntimeException(String.format("Error setting balance"
+                    + "of the fruit %s: quantity %d is less than zero", fruit, quantity));
+        }
+        storage.add(fruit, quantity);
+    }
```

Comment:
The `handle` method should not directly interact with `Storage`. Instead, it should use a service layer to interact with the storage. This will adhere to the SOLID principles, particularly the Single Responsibility Principle and Dependency Inversion Principle.